src/test/validation_tests.cpp

```diff
@@ -57,15 +57,31 @@ BOOST_AUTO_TEST_CASE(subsidy_limit_test)
 {
     const auto chainParams = CreateChainParams(*m_node.args, ChainType::MAIN);
     CAmount nSum = 0;
-    for (int nHeight = 0; nHeight < 14000000; nHeight += 1000) {
+    auto step = 1000;
+    assert(chainParams->GetConsensus().nSubsidyHalvingInterval % step == 0);
+    for (int nHeight = 0; nHeight < 14000000; nHeight += step) {
         CAmount nSubsidy = GetBlockSubsidy(nHeight, chainParams->GetConsensus());
         BOOST_CHECK(nSubsidy <= 50 * COIN);
-        nSum += nSubsidy * 1000;
+        nSum += nSubsidy * step;
         BOOST_CHECK(MoneyRange(nSum));
     }
     BOOST_CHECK_EQUAL(nSum, CAmount{2099999997690000});
 }
 
+// Similar to subsidy_limit_test, but sums every single block subsidy until it drops to zero
+BOOST_AUTO_TEST_CASE(subsidy_sum_test)
+{
+    auto main_consensus{CreateChainParams(*m_node.args, ChainType::MAIN)->GetConsensus()};
+    CAmount total_subsidy_sats{0};
+    int block_height{0};
+    while (auto subsidy = GetBlockSubsidy(block_height, main_consensus)) {
+        total_subsidy_sats += subsidy;
+        block_height++;
+    }
+    BOOST_CHECK_EQUAL(total_subsidy_sats, CAmount{2'099'999'997'690'000});
```

Comment:
Since 21 million BTC is such a meme and [you expressed intent](https://github.com/bitcoin/bitcoin/pull/30699#discussion_r1727707662) to make the test more accessible to novices, I still think this number format could be made clearer. You didn't like my suggestion with the sats-part having `'` after the fourth digit, so this is my final suggestion on the number format:
```suggestion
    BOOST_CHECK_EQUAL(total_subsidy_sats, CAmount{20'999'999'97690000});
```

Reply:
> so this is my final suggestion on the number format

perfect, it's the best of both worlds!